sdk/storage/src/datalake/file_client.cpp

```diff
@@ -0,0 +1,287 @@
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<SharedKeyPolicy>(credential));
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  FileClient::FileClient(
+      const std::string& pathUri,
+      std::shared_ptr<TokenCredential> credential,
+      const FileClientOptions& options)
+      : PathClient(pathUri, credential, options)
+  {
+    Details::InitializeUrisFromServiceUri(pathUri, m_dfsUri, m_blobUri);
+
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.PerOperationPolicies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    // TODO: Retry policy goes here
+    for (const auto& p : options.PerRetryPolicies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<TokenCredentialPolicy>(credential));
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  FileClient::FileClient(const std::string& pathUri, const FileClientOptions& options)
+      : PathClient(pathUri, options)
+  {
+    Details::InitializeUrisFromServiceUri(pathUri, m_dfsUri, m_blobUri);
+
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.PerOperationPolicies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    // TODO: Retry policy goes here
+    for (const auto& p : options.PerRetryPolicies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  PathAppendDataResponse FileClient::AppendData(
+      Azure::Core::Http::BodyStream* content,
+      int64_t offset,
+      const PathAppendDataOptions& options) const
+  {
+    DataLakeRestClient::Path::AppendDataOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Position = offset;
+    protocolLayerOptions.ContentLength = content->Length();
+    protocolLayerOptions.TransactionalContentMD5 = options.ContentMD5;
+    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
```

Comment:
Is there a `LeaseIdMandatory`? If not, why not just call it `LeaseId`?

Reply:
Swagger calls it LeaseIdOptional, however, this is not exposed to customer so I don't think it's bad.

Reply:
I still think it's bad because of inconsistency. But it's okay if you don't want to fix it.